stl/inc/complex

```diff
@@ -990,9 +990,26 @@ protected:
                 this->_Val[_IM] = (this->_Val[_IM] - this->_Val[_RE] * _Wr) / _Wd;
                 this->_Val[_RE] = _Tmp;
             }
-        } else if (_Rightimag == 0) { // set NaN result
-            this->_Val[_RE] = _Myctraits::_Nanv();
-            this->_Val[_IM] = this->_Val[_RE];
+        } else if (_Rightimag == 0) { // _Right.real() == 0 && _Right.imag() == 0
+            if (_Myctraits::_Isnan(this->_Val[_RE])) {
+                this->_Val[_RE] = _Myctraits::_Nanv();
+            } else if (this->_Val[_RE] > 0) {
+                this->_Val[_RE] = _Myctraits::_Infv();
+            } else if (this->_Val[_RE] == 0) {
+                this->_Val[_RE] = -_Myctraits::_Nanv();
+            } else { // this->_Val[_RE] < 0
+                this->_Val[_RE] = -_Myctraits::_Infv();
+            }
+
+            if (_Myctraits::_Isnan(this->_Val[_IM])) {
+                this->_Val[_IM] = _Myctraits::_Nanv();
+            } else if (this->_Val[_IM] > 0) {
+                this->_Val[_IM] = _Myctraits::_Infv();
+            } else if (this->_Val[_IM] == 0) {
+                this->_Val[_IM] = -_Myctraits::_Nanv();
+            } else { // this->_Val[_IM] < 0
+                this->_Val[_IM] = -_Myctraits::_Infv();
+            }

```

Comment:
```suggestion
            this->_Val[_RE] /= _Rightreal;
            this->_Val[_IM] /= _Rightreal;
```
There are two different zeroes in (real) IEEE-754 binary floating point, +0 and -0. Under the default mode, _x_/(-_y_) = -(_x_/_y_) is preserved even when _y_ is zero and _x_ is nonzero: +1/+0 = -1/-0 = +∞, -1/+0 = +1/-0 = -∞.

Each of the real and imaginary parts of a `complex` zero can be either +0 or -0. As a result, there are four different zeros in `complex`, (+0,+0), (+0,-0), (-0,+0) and (-0,-0). It would be nice that complex division by zero also preserves _z_/(-_w_) = -(_z_/_w_): _z_/(-0,+0) = -(_z_/(+0,-0)), _z_/(-0,-0) = -(_z_/(+0,+0)). Dividing by `_Rightreal` is one of the possible ways to achieve this.

Additionally, division by zero raises `FE_DIVBYZERO` or `FE_INVALID` floating-point exception, but `_Ctraits::_Infv()` or `_Ctraits::_Nanv()` doesn't. (https://godbolt.org/z/xzac4G4j8)

ISO C [G.5.1]/5
> If both operands of the `*` operator are complex or if the second operand of the `/` operator is complex, the operator raises floating-point exceptions if appropriate for the calculation of the parts of the result, and may raise spurious floating-point exceptions.